### DISCUSSION TITLE
Is there a way to replace the text in "entry" in the "inputbar" with the selected item?

### DISCUSSION THREAD

**Discussion by Prayag2** [davatorium/rofi#1757] [Q&A]
I use rofi-calc and I would like to replace the text in "entry" (the place where we type) with the result on pressing enter. I've seen [this](https://github.com/svenstaro/rofi-calc/issues/70) but it doesn't work like intended. I want to do exactly what Krunner does. In krunner, on doing calculations and pressing enter, the text in the "input field" is replaced with the result so I can continue with more calculations. How can I achieve this result?

In rofi, on pressing enter, it just closes. I want it to stay and I want xdotool to replace the text in the input field with the result. Please help!

**Comment by DaveDavenport** [comment]
This is something they can easily add  in the plugin, rofi plugin api supports doing this.

(In the result callback, it passes in a `**input`, this is a pointer to the input string that can free'ed and updates with a new value)
